scripts/backup/run_backup/README_HOWTO.txt

```diff
@@ -0,0 +1,103 @@
+
+The following config.ini sections must be configured for the
+whole thing to work:
+
+1. Database. 
+
+The script needs to be able to access the Dataverse database, in order to
+obtain the lists of files that have changed since the last backup and
+need to be copied. The script can use PostgresQL running on a
+remote server. Just make sure that the remote server is configured to
+allow connections from the host running the backup script; and that
+PostgresQL is allowing database access from this host too.
+
+Configure the access credentials as in the example below:
+
+[Database]
+Host: localhost
+Port: 5432
+Database: dvndb
+Username: dvnapp
+Password: xxxxx
+
+In addition to the main Dataverse database, the script maintains its
+own database for keeping track of the backup status of individual
+files. The name of the database is specified in the following setting:
+
+BackupDatabase: backupdb
+
+The database must be created prior to running of the script. For
+example, on the command line: 
+	 createdb -U postgres backupdb --owner=dvnapp
+
+NOTE that the current assumption is that this Postgres database lives
+on the same server as the main Dataverse database and is owned by the
+same user.
+
+Also, one table must be created in this database before the script can
+be run. The script backupdb.sql is provided in this directory. NOTE
+that the Postgres user name dvnapp is hard-coded in the script; change
+it to reflect the name of the database user on your system, if
+necessary.
+
+2. Repository
+
+This section configures access to the datafiles stored in your
+Dataverse. In its present form, the script can read files stored on
+the filesystem and S3. There is no support for reading files stored
+via swift as of yet. Adding swift support should be straightforward,
+by supplying another storage module - similarly to the existing
+storage_filesystem.py and storage_s3.py.
+
+For the filesystem storage: the assumption is that the script has
+direct access to the filesystem where the files live. Meaning that in
+order for the script to work on a server that's different from the one
+running the Dataverse application, the filesystem must be readable by
+the server via NFS, or similarly shared with it.
+
+The filesystem access requires the single configuration setting, as in
+the example below:
+
+[Repository]
+FileSystemDirectory: /usr/local/glassfish4/glassfish/domains/domain1/files
+
+For the S3, no configuration is needed in the config.ini. But AWS
+access must be properly configured for the user running the backup
+module, in the standard ~/.aws location.
+
+
+3. Backup section. 
+
+This section specifies the method for storing the files on the remote
+("secondary") storage subsystem:
+
+[Backup]
+StorageType: ssh
+
+The currently supported methods are "ssh" (the files are transferred
+to the remote location via SSH/SFTP) and "swift" (untested, and
+possibly incomplete implementation is provided; see
+README_IMPLEMENTATION.txt for more details).
```

Comment:
S3 still does not support rsync natively; so the only way I could think of we could use rsync was to download the files from S3 onto the local filesystem, then run rsync on these local copies... not pretty, plus we are going to be somewhat short on local disk space. With sftp, I'm opening an input stream on an S3 object, and write it to the SFTP outputstream in real time.

Reply:
the AWS CLI has a `sync` operation for S3; but I'm not sure if that would be helpful for this

Reply:
Based on what I got reading their docs, the aws tool can behave rsync-like, syncing some local files to S3; (we are planning to use it to sync our production holdings up to S3 during the production migration). But I don't think it can be used to sync some files on S3 up to a remote rsync server.